### PR TITLE
Addition of a new "compact" output format

### LINKED ISSUE #967: Improve console output formatting.
The current console error/info output feels messy and not very readable (even with code snippets turned off) if you have largish number of errors.

It would be nice if it was possible to get a compact output with errors groups by file like `phpstan` or somewhat tabular display like `phpcs`.

### PR DESCRIPTION
**Pull request by erunion**
I've slightly refactored the output formatting system so that it's easier to add new formats, or maintain existing ones, while also adding a new `compact` format. This new format returns your report within a table structure similar to PHPCS.

Example:

```
$ ./psalm --find-dead-code --output-format=compact
FILE: src/Psalm/Internal/FileManipulation/FunctionDocblockManipulator.php

+----------+------+----------------+------------------------------------------------------------------------+
| SEVERITY | LINE | ISSUE          | DESCRIPTION                                                            |
+----------+------+----------------+------------------------------------------------------------------------+
| INFO     | 63   | UnusedProperty | Property Psalm\Internal\FileManipulation\FunctionDocblockManipulator:: |
|          |      |                | $param_typehint_area_starts is never used                              |
| INFO     | 66   | UnusedProperty | Property Psalm\Internal\FileManipulation\FunctionDocblockManipulator:: |
|          |      |                | $param_typehint_starts is never used                                   |
| INFO     | 69   | UnusedProperty | Property Psalm\Internal\FileManipulation\FunctionDocblockManipulator:: |
|          |      |                | $param_typehint_ends is never used                                     |
+----------+------+----------------+------------------------------------------------------------------------+

FILE: src/Psalm/Type/Atomic/TNonEmptyArray.php

+----------+------+------------------------+------------------------------------------------------------------------+
| SEVERITY | LINE | ISSUE                  | DESCRIPTION                                                            |
+----------+------+------------------------+------------------------------------------------------------------------+
| INFO     | 20   | PossiblyUnusedProperty | Cannot find uses of public property Psalm\Type\Atomic\TNonEmptyArray:: |
|          |      |                        | $value                                                                 |
+----------+------+------------------------+------------------------------------------------------------------------+
```

It also, like the default `console` format, supports colors if errors are present:

screen shot 2018-12-09 at 3 54 11 pm

Resolves #967

### CODE REVIEW

File: src/Psalm/Output/Compact.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                if ($buffer !== null) {
+                    $table->render();
+                    $output[] = $buffer->fetch();
+                }
+
+                $output[] = 'FILE: ' . $issue_data['file_name'] . "\n";
+
+                $buffer = new BufferedOutput();
+                $table = new Table($buffer);
+                $table->setColumnMaxWidth(3, 70);
```
Comment (erunion) [inline src/Psalm/Output/Compact.php]: Looks like this isn't available in symfony/console 3.x

Comment (weirdan) [reply src/Psalm/Output/Compact.php]: Available since 4.2

Comment (erunion) [reply src/Psalm/Output/Compact.php]: Replaced with an alternate wrapping implementation.

**Comment by muglug** [comment]
This is awesome - just waiting on a clean bill of health from Travis, then I'll merge

**Comment by muglug** [comment]
Thanks!

**Pull request merged**